kratos/utilities/read_materials_utility.cpp

```diff
@@ -245,32 +287,177 @@ void ReadMaterialsUtility::AssignPropertyBlock(Parameters Data)
+                // We get the adress if any
+                const std::string& r_use_existing_property = sub_prop["use_existing_property"].GetString();
+
+                // We check if already defined
+                bool already_defined = false;
+                if (r_use_existing_property != "") { // NOTE: This means that is not marked as existing
+                    if (r_use_existing_property.size() > 1) {
+                        if (rModelPart.HasProperties(r_use_existing_property, mesh_id)) {
+                            p_new_sub_prop = rModelPart.pGetProperties(r_use_existing_property, mesh_id);
+                            already_defined = true;
+                        }
+                    } else {
+                        const IndexType property_id = std::stoi(r_use_existing_property);
+                        if (rModelPart.RecursivelyHasProperties(property_id)) {
+                            p_new_sub_prop = rModelPart.pGetProperties(property_id);
+                            already_defined = true;
+                        }
+                    }
+                }
+
+                // Check if properly read use_existing_property
+                KRATOS_ERROR_IF_NOT(already_defined) << "Subproperties " << r_use_existing_property << " is not already defined. You need to check the structure of your materials file" << std::endl;
+            } else { // We get or create the new subproperty
+                // We get the subproperty id
+                const int sub_property_id = sub_prop["properties_id"].GetInt();
+
+                // Actually creating it
+                p_new_sub_prop = rModelPart.HasProperties(sub_property_id, mesh_id) ? rModelPart.pGetProperties(sub_property_id, mesh_id) : rModelPart.CreateNewProperties(sub_property_id, mesh_id);
+
+                // Read the recursively subproperties
+                CreateSubProperties(rModelPart, sub_prop, p_new_sub_prop);
+
+                // We create the new sub property
+                if (sub_prop.Has("Material")) {
+                    CreateProperty(sub_prop, p_new_sub_prop, true);
+                }
+            }
+
+            r_list_sub_properties.insert(r_list_sub_properties.begin(), p_new_sub_prop);
+        }
+    }
+
+    KRATOS_CATCH("");
+}
+
+/***********************************************************************************/
+/***********************************************************************************/
+
+void ReadMaterialsUtility::AssignPropertyBlock(Parameters Data)
+{
+    KRATOS_TRY;
+
+    // Get the properties for the specified model part.
+    ModelPart& r_model_part = mrModel.GetModelPart(Data["model_part_name"].GetString());
+    const IndexType property_id = Data["properties_id"].GetInt();
+    const IndexType mesh_id = 0;
+    Properties::Pointer p_prop;
+    if (r_model_part.RecursivelyHasProperties(property_id, mesh_id)) {
+        KRATOS_WARNING("ReadMaterialsUtility") << "WARNING:: The properties ID: " << property_id
```

Comment:
I believe this warning has t o be moved to `GetPropertyBlock`, since there you create the (main-)properties the first time, then you call `AssignPropertyBlock` => hence the Properties will always exist and this warning will wrongly be issued